core/txpool/legacypool/list.go

```diff
@@ -160,13 +160,11 @@ func (m *sortedMap) Cap(threshold int) types.Transactions {
 	// Otherwise gather and drop the highest nonce'd transactions
 	var drops types.Transactions
 
-	sort.Sort(*m.index)
 	for size := len(m.items); size > threshold; size-- {
-		drops = append(drops, m.items[(*m.index)[size-1]])
-		delete(m.items, (*m.index)[size-1])
+		dropIdx := m.index.Pop().(uint64)
```

Comment:
The documentation of container/heap says: 

> Note that [Push](https://pkg.go.dev/container/heap#Push) and [Pop](https://pkg.go.dev/container/heap#Pop) in this interface are for package heap's implementation to call. To add and remove things from the heap, use [heap.Push](https://pkg.go.dev/container/heap#Push) and [heap.Pop](https://pkg.go.dev/container/heap#Pop).

So you need to call `heap.Pop(m.index)` here.